### DISCUSSION TITLE
baseBranches docs unclear

### DISCUSSION THREAD

**Discussion by MShekow** [renovatebot/renovate#10510] [Request Help] [answered]
Hi. The steps done by the bot in case `baseBranches` is set to an array of branches is unclear (from just reading the docs).

From what I understand, Renovate bot will always first visit the repo's *default* branch. If the `renovate.json` config of the *default* branch contains something like `"baseBranches": ["master", "foo", "bar"]`, the bot will also scan for outdated dependencies/packages in `foo` and `bar`. However, the following things are unclear:

- Do I need to have a `renovate.json` config in these branches (`foo` and `bar`)?
- If not, will the bot just use the configuration from the default branch? Or would it initiate an onboarding PR?
- If there is a `renovate.json` config in these branches (`foo` and `bar`), with configuration options that differ from those in the *default* branch, how will the bot behave? Are these merged? Or completely overwritten? What happens if `renovate.json` config in the `foo` branch contains something like `"baseBranches": ["master", "foo", "fish"]`?

**Comment by viceice** [comment, marked as answer]
> 
> * Do I need to have a `renovate.json` config in these branches (`foo` and `bar`)?

Nope, renovate will (currently) only use the config from default branch. see #2496

> * If not, will the bot just use the configuration from the default branch? Or would it initiate an onboarding PR?

It will onboard if there is no existing onboarding pr and no config in default branch.

> * If there is a `renovate.json` config in these branches (`foo` and `bar`), with configuration options that differ from those in the _default_ branch, how will the bot behave? Are these merged? Or completely overwritten? What happens if `renovate.json` config in the `foo` branch contains something like `"baseBranches": ["master", "foo", "fish"]`?

See first answer. You can use `matchBaseBranches` to scope package rules to branches

https://docs.renovatebot.com/configuration-options/#matchbasebranches
